src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/controller/cas2bail/Cas2BailApplicationController.kt

```diff
@@ -0,0 +1,175 @@
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.problem.ForbiddenProblem
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.problem.NotFoundProblem
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.results.AuthorisableActionResult
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.results.ValidatableActionResult
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.service.HttpAuthService
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.service.NomisUserService
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.service.cas2.OffenderService
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.service.cas2bail.Cas2BailApplicationService
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.transformer.cas2bail.Cas2BailApplicationsTransformer
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.util.PageCriteria
+import java.net.URI
+import java.util.*
+
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.api.model.Cas2ApplicationSummary as ModelCas2ApplicationSummary
+
+@Service(
+  "uk.gov.justice.digital.hmpps.approvedpremisesapi.controller.cas2bail" +
+    ".Cas2BailApplicationsController",
+)
+class Cas2BailApplicationController(
+  private val httpAuthService: HttpAuthService,
+  private val cas2BailApplicationService: Cas2BailApplicationService,
+  private val cas2BailApplicationsTransformer: Cas2BailApplicationsTransformer,
+  private val objectMapper: ObjectMapper,
+  private val offenderService: OffenderService,
+  private val userService: NomisUserService,
+) : ApplicationsCas2bailDelegate {
+
+  override fun applicationsGet(
+    isSubmitted: Boolean?,
+    page: Int?,
+    prisonCode: String?,
+  ): ResponseEntity<List<ModelCas2ApplicationSummary>> {
+    val user = userService.getUserForRequest()
+
+    prisonCode?.let { if (prisonCode != user.activeCaseloadId) throw ForbiddenProblem() }
+
+    val pageCriteria = PageCriteria("createdAt", SortDirection.desc, page)
+
+    val (applications, metadata) = cas2BailApplicationService.getCas2BailApplications(prisonCode, isSubmitted, user, pageCriteria)
+
+    return ResponseEntity.ok().headers(
+      metadata?.toHeaders(),
+    ).body(getPersonNamesAndTransformToSummaries(applications))
+  }
+
+  override fun applicationsApplicationIdGet(applicationId: UUID): ResponseEntity<Application> {
+    val user = userService.getUserForRequest()
+
+    val application = when (
+      val applicationResult = cas2BailApplicationService
+        .getCas2BailApplicationForUser(
+          applicationId,
+          user
+        )
+
+    ) {
+      is AuthorisableActionResult.NotFound -> null
+      is AuthorisableActionResult.Unauthorised -> throw ForbiddenProblem()
+      is AuthorisableActionResult.Success -> applicationResult.entity
+    }
+
+    if (application != null) {
+      return ResponseEntity.ok(getPersonDetailAndTransform(application))
+    }
+    throw NotFoundProblem(applicationId, "Application")
```

Comment:
I appreciate that you've copied this code, but this NotFoundProblem can be moved into line 70